File: cs50/problems/2025/x/sentimental/mario/less/mario.py
from cs50 import get_int

while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break


for i in range(height):
    for j in range(height):
        spaces = height - 1 - i
        segments = i + 1
    print(" " * spaces, end="")
    print("#" * segments, end="")
    print()

File: cs50/problems/2025/x/mario/more/mario.c
#include <cs50.h>
#include <stdio.h>

void draw_level(int height);
void draw_line(int spaces, int bricks);

int main(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    draw_level(height);
}

void draw_level(int height)
{
    for (int i = 1; i <= height; i++)
    {
        int spaces = height - i;
        int bricks = i;

        draw_line(spaces, bricks);
    }
}

void draw_line(int spaces, int bricks)
{
    for (int i = 0; i < spaces; i++)
    {
        printf(" ");
    }

    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }

    printf("  ");

    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }

    printf("\n");
}
